torch_geometric/explain/explanation.py

```diff
@@ -129,3 +131,172 @@ def _apply_masks(
             out = out.subgraph(node_mask)
 
         return out
+
+
+class HeteroExplanation(HeteroData):
+    r"""Holds all the obtained explanations of a heterogeneous graph.
+
+    The explanation object is a :obj:`~torch_geometric.hetero_data.HeteroData`
+    object and can hold node-attribution, edge-attribution,
+    feature-attribution. It can also hold the original graph if needed.
+
+    Args:
+        node_mask (Dict[NodeType, Tensor], optional): Dictionary of Node-level
+            mask with  key `NodeType` and value `Tensor` of
+            shape :obj:`[num_nodes]`. (default: :obj:`None`)
+        edge_mask (Dict[EdgeType, Tensor], optional): Dictionary of Edge-level
+            mask with  key `EdgeType` and value `Tensor` of
+            shape :obj:`[num_edges]`. (default: :obj:`None`)
+        node_feat_mask (Dict[NodeType, Tensor], optional): Dictionary of
+            Node-level feature mask with  key `NodeType` and value `Tensor` of
+            shape :obj:`[num_nodes, num_node_features]`. (default: :obj:`None`)
+        edge_feat_mask (Dict[EdgeType, Tensor], optional): Dictionary of
+            Edge-level feature mask with  key `EdgeType` and value `Tensor` of
+            shape :obj:`[num_edges, num_edge_features]`. (default: :obj:`None`)
+        **kwargs (optional): Additional attributes.
+    """
+    def __init__(
+        self,
+        node_mask: Optional[Dict[NodeType, Tensor]] = None,
+        edge_mask: Optional[Dict[EdgeType, Tensor]] = None,
+        node_feat_mask: Optional[Dict[NodeType, Tensor]] = None,
+        edge_feat_mask: Optional[Dict[EdgeType, Tensor]] = None,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+
+        # HeteroData doesn't set all kwargs as attributes, so we need to do it
```

Comment:
```suggestion
    def __init__(
        x_dict, edge_index_dict, edge_attr_dict = None,
        node_mask: Optional[Dict[NodeType, Tensor]] = None,
        edge_mask: Optional[Dict[EdgeType, Tensor]] = None,
        node_feat_mask: Optional[Dict[NodeType, Tensor]] = None,
        edge_feat_mask: Optional[Dict[EdgeType, Tensor]] = None,
    ):
        super().__init__()
        for key, val in x_dict.items():
            self[key].x = val
        for key, val in edge_index_dict.items():
            self[key].edge_index = val
            if edge_attr_dict is not None:
                self.[key].edge_attr = edge_attr_dict[key]
        # HeteroData doesn't set all kwargs as attributes, so we need to do it
```
Assigning it this way should help you create a valid `HeteData` object on which `subgraph` will work. Note, I made the assumption that `x_dict` and `edge_index_dict` are passed as arguments, which is reasonable given that we create `HeteroExplanation` internally.

Reply:
Thanks @wsad1! Resolving this comment as @dufourc1's comment is also below.